todo_app/tests/test_app.py

```diff
@@ -0,0 +1,69 @@
+import pytest, requests, os
+from dotenv import load_dotenv, find_dotenv
+from todo_app import app
+import vcr
+import urllib.request
+
+@pytest.fixture
+def client():
+    file_path = find_dotenv('.env')
+    load_dotenv(file_path, override=True)
+
+    test_app = app.create_app()
+
+    with test_app.test_client() as client:
+        yield client
+
+class StubResponse():
+    def __init__(self, fake_response_data):
+        self.fake_response_data = fake_response_data
+
+    def json(self):
+        return self.fake_response_data
+
+def stub(method, url, params={}):
+    test_board_id = os.environ.get('TRELLO_BOARD_ID')
+    trello_api_key = os.environ.get('TRELLO_API_KEY')
+    trello_api_token = os.environ.get('TRELLO_API_TOKEN')
+    trello_board_id = os.environ.get('TRELLO_BOARD_ID')
+
+    if url == f'https://api.trello.com/1/boards/{test_board_id}/lists' and method == "GET":
+        with vcr.use_cassette('fixtures/vcr_cassettes/synopsis.yaml'):
+            vcr_response_data = urllib.request.urlopen(f'https://api.trello.com/1/boards/{trello_board_id}/lists?key={trello_api_key}&token={trello_api_token}&cards=open').read()
```

Comment:
This seems to return a byte array, which means it's totally the wrong format when our app code tries to convert to a list/dictionary with the `.json()` function call.

To get this working, I had to adjust to:
```python
        with vcr.use_cassette('fixtures/vcr_cassettes/synopsis.yaml', filter_query_parameters=['key', 'token']):
            resource = urllib.request.urlopen(f'https://api.trello.com/1/boards/{trello_board_id}/lists?key={trello_api_key}&token={trello_api_token}&cards=open')
            vcr_response_data = resource.read().decode('utf-8')
        return StubResponse(json.loads(vcr_response_data))
```

Reply:
Note the `filter_query_parameters` argument as well, which is great for automatically stripping sensitive values!